server/src/utils/fight/fightMethods.ts

```diff
@@ -1472,11 +1553,11 @@ const evade = (fighter: DetailedFighter, opponent: DetailedFighter, difficulty =
 
   return random * difficulty
     < Math.min(
-      (getFighterStat(opponent, 'evasion')
+      (opponentEvasion
         + agilityDifference * 0.01
         - getFighterStat(fighter, 'accuracy')
         - getFighterStat(fighter, 'dexterity')),
-      0.9,
+      0.9 * difficulty,
```

Comment:
Why?

Reply:
Because let's say a brute has 160% evasion, without this formula, evasion on ranged attacks is capped at 45%, which is quite unfair...

The only point of the 0.9, is to prevent never ending fights, which it also does in the new formula.

To summarize : without the formula, evasion is capped at 90% for melee and 45% for ranged, with the formula, evasion is capped at 90% for melee and 90% for ranged, which is better.